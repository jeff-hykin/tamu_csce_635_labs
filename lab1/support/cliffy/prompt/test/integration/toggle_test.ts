import { Toggle } from "../../toggle.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";

await snapshotTest({
  name: "toggle prompt",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should toggle the prompt": { stdin: ["y", "n", "y", "\n"] },
  },
  async fn() {
    await Toggle.prompt({
      message: "Please confirm",
      hint: "some hint",
      default: false,
    });
  },
});
