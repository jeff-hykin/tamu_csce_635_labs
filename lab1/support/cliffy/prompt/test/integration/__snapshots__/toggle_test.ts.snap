export const snapshot = {};

snapshot[`toggle prompt > should toggle the prompt 1`] = `
stdout:
"? Please confirm (No) › No / Yes
❯ some hint\\x1b[1A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Please confirm (No) › No / Yes
❯ some hint\\x1b[1A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Please confirm (No) › No / Yes
❯ some hint\\x1b[1A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Please confirm (No) › No / Yes
❯ some hint\\x1b[1A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Please confirm (No) › Yes
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
